### PR TITLE
implement endian-aware reader support

### PR DESCRIPTION
**Pull request by hobu**

### CODE REVIEW

File: pdal/io/NumpyReader.cpp [hunk 1]
```diff
@@ -530,12 +540,101 @@ bool NumpyReader::nextPoint()
 }
 
 
+
+
 bool NumpyReader::loadPoint(PointRef& point, point_count_t position)
 {
     using namespace Dimension;
 
+    pdal::SwitchableExtractor extractor(p_data, *m_strideptr);
```
[the remark this answers is not in the document]
Comment (hobu) [reply pdal/io/NumpyReader.cpp:549]: Then I would need to make an extractor for each field. It is possible that each field in the numpy array has different endianness.

File: pdal/io/NumpyReader.cpp [hunk 2]
```diff
@@ -530,12 +540,101 @@ bool NumpyReader::nextPoint()
...
+
     for (const Field& f : m_fields)
-        point.setField(f.m_id, f.m_type, (void*)(p_data + f.m_offset));
+    {
+        if (f.m_byteorder == '>')
+            extractor.switchToBigEndian();
+        else
+            extractor.switchToLittleEndian();
+
+        if (Dimension::base(f.m_type) == Dimension::BaseType::Floating)
```
[the remark this answers is not in the document]
Comment (abellgithub) [reply pdal/io/NumpyReader.cpp:571]: Sorry this seems messy.

File: pdal/io/NumpyReader.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (abellgithub) [reply pdal/io/NumpyReader.cpp:549]: No.  The byte order is fixed.

File: pdal/io/NumpyReader.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (hobu) [reply pdal/io/NumpyReader.cpp:571]: I'll switch it to a switch statement as described.

**Pull request merged**